### DISCUSSION TITLE
Why limit the user to a minimum size of the window

### DISCUSSION THREAD

**Discussion by DreamMaoMao** [blueman-project/blueman#2164] [General]
In the non-floating state
On some tiled window managers in the wayland environment blueman will refuse to shrink to a smaller size than its window limit minimum

In the floating state
In both waylabd and x11 environments it refuses to shrink to a smaller size than its windowed minimum

This is not available in other applications, so if you want to use it, you can only accept its limited window size and no way to customize it

**Comment by infirit** [comment]
Iirc we only enforce the horizontal scroll policy (no scrolling) on the DeviceList. You should be able to make the vertical size quite small.

**Comment by DreamMaoMao** [comment]
![image](https://github.com/blueman-project/blueman/assets/30348075/c023ad3a-45f2-41f9-b2e2-40ce12b0e0ad)

@infirit 

There are some excess areas on Horizontal direction  I think it is not automatically in an optimal size, is it possilbe to allow users to adjust it themselves

**Comment by DreamMaoMao** [comment]
![image](https://github.com/blueman-project/blueman/assets/30348075/8a785b51-9557-4cb7-bf93-dbbf5c51fda3)

When I sent a resize request to the blueman window in the x11 non-floating state, he could accept it and reset his size, but when I tried to send a smaller request to him in the wayland or floating state, the blueman window refused to do it

**Comment by infirit** [comment]
We set a property on a widget which forces a specific width. You can try #2165 which changes this.

It does have a drawback, the progressbar can cause the window to resize as it needs some space (it is hidden on launch).

edit: 200px seems enough space for the progressbar.

**Comment by DreamMaoMao** [comment]
The x11 environment can be shrunk down, but the wayland environment still can't be shrunk down

**Comment by infirit** [comment]
Just checked and I see identical behaviour under kwin wayland and kwin X11. This is the minimum size under kwin wayland.

![image](https://github.com/blueman-project/blueman/assets/3465730/dd911d1f-20c5-4cc0-abef-90d839d89cff)

**Comment by DreamMaoMao** [comment]
Oh thanks for you test . I am using dwl, I still have some problems with the window fit in wayland environment, I gave up letting wayland Windows automatically resize, I tried the more general force-cut application display window to avoid tiling chaos.
